pkg/controller/kogitoinfra/mongodb.go

```diff
@@ -0,0 +1,330 @@
+	envProps = append(envProps, framework.CreateSecretEnvVar(propertiesMongoDBQuarkus[envVarMongoDBDatabase], mongoDBSecretName, infrastructure.MongoDBAppSecretDatabaseKey))
+	return envProps, nil
+}
+
+func (i *mongoDBInfraReconciler) getMongoDBAppProps(mongoDBInstance *mongodb.MongoDB) (map[string]string, error) {
+	appProps := map[string]string{}
+
+	mongoDBURI := mongoDBInstance.Status.MongoURI
+	if len(mongoDBURI) > 0 {
+		mongoDBParsedURL, err := url.ParseRequestURI(mongoDBURI)
+		if err != nil {
+			return nil, err
+		}
+		appProps[propertiesMongoDBSpring[appPropMongoDBHost]] = mongoDBParsedURL.Hostname()
+		appProps[propertiesMongoDBSpring[appPropMongoDBPort]] = mongoDBParsedURL.Port()
+		appProps[propertiesMongoDBQuarkus[appPropMongoDBURI]] = mongoDBURI
+	}
+
+	return appProps, nil
+}
+
+func (i *mongoDBInfraReconciler) updateMongoDBAppPropsInStatus(mongoDBInstance *mongodb.MongoDB) error {
+	log.Debug("going to Update MongoDB app properties in kogito infra instance status")
+	appProps, err := i.getMongoDBAppProps(mongoDBInstance)
+	if err != nil {
+		return err
+	}
+	i.instance.Status.AppProps = appProps
+	log.Debugf("Following app properties are set infra status : %s", appProps)
+	return nil
+}
+
+func (i *mongoDBInfraReconciler) updateMongoDBEnvVarsInStatus(mongoDBInstance *mongodb.MongoDB) error {
+	log.Debug("going to Update MongoDB env properties in kogito infra instance status")
+	envVars, err := i.getMongoDBSecretEnvVars(mongoDBInstance)
+	if err != nil {
+		return err
+	}
+	i.instance.Status.Env = envVars
+	log.Debugf("Following env properties are set infra status : %s", envVars)
+	return nil
+}
+
+func (i *mongoDBInfraReconciler) loadDeployedMongoDBInstance(instanceName string, namespace string) (*mongodb.MongoDB, error) {
+	log.Debug("fetching deployed kogito mongoDB instance")
+	mongoDBInstance := &mongodb.MongoDB{}
+	if exists, err := kubernetes.ResourceC(i.client).FetchWithKey(types.NamespacedName{Name: instanceName, Namespace: namespace}, mongoDBInstance); err != nil {
+		log.Error("Error occurs while fetching kogito mongoDB instance")
+		return nil, err
+	} else if !exists {
+		log.Debug("Kogito mongoDB instance is not exists")
+		return nil, nil
+	} else {
+		log.Debug("Kogito mongoDB instance found")
+		log.Debug("Kogito mongoDB instance found %v", mongoDBInstance)
+		return mongoDBInstance, nil
+	}
+}
+
+func (i *mongoDBInfraReconciler) loadCustomKogitoMongoDBSecret(namespace string) (*corev1.Secret, error) {
+	log.Debugf("Fetching %s ", mongoDBSecretName)
+	secret := &corev1.Secret{}
+	if exits, err := kubernetes.ResourceC(i.client).FetchWithKey(types.NamespacedName{Name: mongoDBSecretName, Namespace: namespace}, secret); err != nil {
+		log.Errorf("Error occurs while fetching %s", mongoDBSecretName)
+		return nil, err
+	} else if !exits {
+		log.Errorf("%s not found", mongoDBSecretName)
```

Comment:
Does this need to be logged as an error (or logged at all)?
Missing custom Kogito secret can mean that the operator just didn't create a new one yet. We don't have a broader context in this function so IMHO it should be ok to return nil and not log anything. Possible logging can be handled by other functions using this function.

Reply:
Or possible reducing the log level to `Debugf`